src/__tests__/variables.test-d.ts

```diff
@@ -36,6 +36,22 @@ describe('getVariablesType', () => {
     }>();
   });
 
+  it('works for one-of-objects', () => {
+    const query = `
+      mutation ($id: ID!, $input: OneOfPayload!) {
+        toggleTodo (id: $id input: $input) { id }
+      }
+    `;
+
+    type doc = parseDocument<typeof query>;
+    type variables = getVariablesType<doc, schema>;
+
+    expectTypeOf<variables>().toEqualTypeOf<{
+      id: string;
+      input: { value_1: string } | { value_2: string };
```

Comment:
Just discovered a bit of a wrench in the current implementation:
> Within the input object literal or unordered map, if the single entry is {null} an error must be thrown.

In the example that this PR provides, all input fields are defined as non-nullable, however, in the example that the RFC provides all fields are nullable. Meaning, it's irrelevant whether a field is non-nullable or nullable.

It seems like all input fields must be non-null.